wall/wall.c

```diff
@@ -0,0 +1,125 @@
+/*
+ * wall.c - write to all users
+ *
+ * Copyright (C) 2024: Luiz Antônio Rangel (takusuman)
+ *
+ * SPDX-Licence-Identifier: Zlib
+ *
+ *	from Unix 32V /usr/src/cmd/wall.c
+ *	December 19th, 1978.
+ *
+ * Copyright(C) Caldera International Inc. 2001-2002. All rights reserved.
+ *
+ * SPDX-Licence-Identifier: Caldera
+ */
+
+#include <errno.h>
+#include <limits.h>
+#include <pwd.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <strmenta.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <utmp.h>
+#include <utmpx.h>
+#define MAXNAMLEN _POSIX_LOGIN_NAME_MAX
+
+char	*mesg = "",
+	who[MAXNAMLEN] = "???",
+	sterm[32] = "";
+int	msize = 0;
+void main(int argc, char *argv[]);
+void sendmes(char *tty);
+
+void main(int argc, char *argv[]) {
+	int i = 0;
+	char c = '\0',
+	     smesg[3000] = "";
+	struct utmpx *utmp;
+	struct passwd *pw;
+	FILE *f = NULL;
+
+	if ((utmp = getutxent()) == NULL) {
+		fprintf(stderr, "failed to open utmpx database: %s\n",
+			strerror(errno));
+		exit(1);
+	}
+
+	f = stdin;
```

Comment:
You can move the `FILE *f = NULL;` declaration out of the top of the function to here and replace the `NULL` by `stdin`

Reply:
That was there since the 70's, I didn't change a lot this code, just ported it to the modern utmp(x) database.

Reply:
If it's ANSI C it should be kept like that since `stdin` can be a macro

Reply:
Nice. Also done.